### PR TITLE
Switch to github.com/vmware-labs/yaml-jsonpath/pkg/yamlpath for jsonpath

### PR DESCRIPTION
**Pull request by marwatk**
Old library didn't support filtering on child nodes (e.g. `$.images[?(@.publish)].tag`).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,21 +4,23 @@ require (
 	github.com/evanphx/json-patch/v5 v5.6.0
 	github.com/imdario/mergo v0.3.12
 	github.com/mitchellh/mapstructure v1.5.0
-	github.com/ohler55/ojg v1.14.0
+	github.com/ohler55/ojg v1.14.3
```
Comment (lucastheisen) [inline go.mod]: Why is this library still included?

Comment (marwatk) [reply go.mod]: Need a go mod tidy. Done.

File: pkg/cmd/jsonpath.go [hunk 2]
```diff
@@ -57,6 +52,45 @@ func (c jsonpathContext) jsonpath(
 	return nil
 }
 
+func evaluateJSONPath(path string, data interface{}, first bool) (interface{}, error) {
```
Comment (lucastheisen) [inline pkg/cmd/jsonpath.go:55]: This needs some commentary as to the necessity of the marshal/unmarshal/marshal sequence.

Comment (marwatk) [reply pkg/cmd/jsonpath.go:55]: Added.

**Pull request merged**